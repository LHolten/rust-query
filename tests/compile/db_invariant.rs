use rust_query::ops::Join;

fn main() {}

fn test<'a: 'b, 'b>(val: Join<'a, ()>) -> Join<'b, ()> {
    val
}
