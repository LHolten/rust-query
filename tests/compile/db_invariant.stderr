error: lifetime may not live long enough
 --> tests/compile/db_invariant.rs:6:5
  |
5 | fn test<'a: 'b, 'b>(val: Join<'a, ()>) -> Join<'b, ()> {
  |         --      -- lifetime `'b` defined here
  |         |
  |         lifetime `'a` defined here
6 |     val
  |     ^^^ function was supposed to return data with lifetime `'a` but it is returning data with lifetime `'b`
  |
  = help: consider adding the following bound: `'b: 'a`
  = note: requirement occurs because of the type `Join<'_, ()>`, which makes the generic argument `'_` invariant
  = note: the struct `Join<'t, T>` is invariant over the parameter `'t`
  = help: see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance
